src/a2a/types/extras.py

```diff
@@ -0,0 +1,360 @@
+    result: Any
+    id: str | int | None = None
+
+
+class JSONRPCErrorResponse(A2ABaseModel):
+    """Represents a JSON-RPC 2.0 Error Response object."""
+
+    jsonrpc: Literal["2.0"] = "2.0"
+    error: A2AError
+    id: str | int | None = None
+
+
+# Type alias for A2A requests (union of all request types)
+# This maps to the various request message types in the proto
+from a2a.types.a2a_pb2 import (
+    CancelTaskRequest,
+    GetExtendedAgentCardRequest,
+    GetTaskPushNotificationConfigRequest,
+    GetTaskRequest,
+    SendMessageRequest,
+    SetTaskPushNotificationConfigRequest,
+    SubscribeToTaskRequest,
+)
+
+A2ARequest = Union[
+    SendMessageRequest,
+    GetTaskRequest,
+    CancelTaskRequest,
+    SetTaskPushNotificationConfigRequest,
+    GetTaskPushNotificationConfigRequest,
+    SubscribeToTaskRequest,
+    GetExtendedAgentCardRequest,
+]
+
+
+# JSON-RPC Success Response types
+# These wrap the result of successful RPC calls
+# Note: result is typed as Any to allow both proto messages and dicts
+class GetTaskSuccessResponse(A2ABaseModel):
+    """Success response for GetTask RPC."""
+    jsonrpc: Literal["2.0"] = "2.0"
+    id: str | int | None = None
+    result: Any
+
+
+class CancelTaskSuccessResponse(A2ABaseModel):
+    """Success response for CancelTask RPC."""
+    jsonrpc: Literal["2.0"] = "2.0"
+    id: str | int | None = None
+    result: Any
+
+
+class SendMessageSuccessResponse(A2ABaseModel):
+    """Success response for SendMessage RPC."""
+    jsonrpc: Literal["2.0"] = "2.0"
+    id: str | int | None = None
+    result: Any
+
+
+class SendStreamingMessageSuccessResponse(A2ABaseModel):
+    """Success response for streaming message RPC."""
+    jsonrpc: Literal["2.0"] = "2.0"
+    id: str | int | None = None
+    result: Any  # Streaming events
+
+
+class SetTaskPushNotificationConfigSuccessResponse(A2ABaseModel):
+    """Success response for SetTaskPushNotificationConfig RPC."""
+    jsonrpc: Literal["2.0"] = "2.0"
+    id: str | int | None = None
+    result: Any
+
+
+class GetTaskPushNotificationConfigSuccessResponse(A2ABaseModel):
+    """Success response for GetTaskPushNotificationConfig RPC."""
+    jsonrpc: Literal["2.0"] = "2.0"
+    id: str | int | None = None
+    result: Any
+
+
+class ListTaskPushNotificationConfigSuccessResponse(A2ABaseModel):
+    """Success response for ListTaskPushNotificationConfig RPC."""
+    jsonrpc: Literal["2.0"] = "2.0"
+    id: str | int | None = None
+    result: Any
+
+
+class DeleteTaskPushNotificationConfigSuccessResponse(A2ABaseModel):
+    """Success response for DeleteTaskPushNotificationConfig RPC."""
+    jsonrpc: Literal["2.0"] = "2.0"
+    id: str | int | None = None
+    result: None = None
+
+
+class GetAuthenticatedExtendedCardSuccessResponse(A2ABaseModel):
+    """Success response for GetAuthenticatedExtendedCard RPC."""
+    jsonrpc: Literal["2.0"] = "2.0"
+    id: str | int | None = None
+    result: Any  # AgentCard
```

Comment:
Let not add our own implementation of this. In the client code I pull in the "jsonrpc" package from pypi, which has the JSONRPC20Response type already defined, then we only need to concern ourselves with populating the result or error fields.